static/js/components/CompactPostDisplay.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+import React from "react"
+import moment from "moment"
+import { connect } from "react-redux"
+import { Link } from "react-router-dom"
+
+import { channelURL, postDetailURL } from "../lib/url"
+import {
+  formatCommentsCount,
+  formatPostTitle,
+  PostVotingButtons
+} from "../lib/posts"
+
+import type { Post } from "../flow/discussionTypes"
+
+class CompactPostDisplay extends React.Component {
+  props: {
+    post: Post,
+    showChannelLink?: boolean,
+    toggleUpvote: Post => void
+  }
+
+  state: {
+    upvoting: boolean
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      upvoting: false
```

Comment:
Should this be part of `PostVotingButtons`?

Reply:
yeah that's a good idea